src/index.ts

```diff
@@ -409,15 +401,15 @@ function visitFunctionLikeDeclaration(node: ts.FunctionLikeDeclaration, kind: ts
     }
 
     let moveOpenParen = false;
+    const params = node.parameters;
     if (node.typeParameters && node.typeParameters.length) {
-        moveOpenParen = isAsync(node.modifiers) && spansLines(node.typeParameters.pos, node.typeParameters.end);
```

Comment:
For some reason when I fixed a similar bug before I only applied the fix for `async` arrows. The `return` case hadn't occurred to me at the time.